### PR TITLE
round out scancode definition sumarization

### PR DESCRIPTION
**Pull request by jeffmcaffer**
update the definition schema and enhance scancode summarization as follows

* ensure props are set only if they have values
* associate all facets with each applicable file
* support license files only in the root
* update and enhance the tests

### CODE REVIEW

File: providers/summary/scancode.js [hunk 1]
```diff
@@ -27,29 +27,28 @@ class ScanCodeSummarizer {
...
     if (facetList.length === 0) return { core: files }
-    const result = {}
+    const result = { core: [...files] }
     for (const facet in facetList) {
       const facetKey = facetList[facet]
       const filters = facets[facetKey]
       if (!filters || filters.length === 0) break
-      result[facetKey] = remove(files, file => filters.some(filter => minimatch(file.path, filter)))
+      result[facetKey] = files.filter(file => filters.some(filter => minimatch(file.path, filter)))
+      pullAllWith(result.core, result[facetKey], isEqual)
```
Comment (Teju-Manchenella) [inline providers/summary/scancode.js]: What is minimatch and how does it work

File: providers/summary/scancode.js [hunk 2]
```diff
@@ -58,79 +57,90 @@ class ScanCodeSummarizer {
...
-        if (!file.licenses || file.licenses.length === 0) unknownLicenses++
-        const hasHolders = this._normalizeCopyrights(file.copyrights, copyrightHolders)
-        !hasHolders && unknownParties++
+    for (let file of facetFiles) {
+      file.license ? licenseExpressions.add(file.license) : unknownLicenses++
+      file.attributions ? this._addArrayToSet(file.attributions, attributions) : unknownParties++
+      if (facet !== 'core') {
+        // tag the file with the current facet if not core
+        file.facets = file.facets || []
+        file.facets.push(facet)
```
Comment (Teju-Manchenella) [inline providers/summary/scancode.js]: What is being added to the file.facets array?

Comment (jeffmcaffer) [reply providers/summary/scancode.js]: the list of facet names that apply to the individual file

File: providers/summary/scancode.js (hunk 1 above)
Comment (jeffmcaffer) [reply providers/summary/scancode.js]: https://www.npmjs.com/package/minimatch

**Pull request merged**